narwhal/node/src/lib.rs

```diff
@@ -229,6 +233,14 @@ impl Node {
             )
         };
 
+        // We check the recovery token here which is always true if the consensus is enabled.
+        // This allows us to ensure that the primary is spawned only after the
+        // consensus is guaranteed to be in a state where it can process messages it
+        // receives from the primary when the primary starts up.
+        if consensus_recovery_token {
+            info!("Consensus is ready for messages, now starting the primary");
+        }
+
```

Comment:
Based on my above comment this won't be needed any more here - I mean as part of our node process we do have a hard requirement for consensus to get restored first, so maybe we don't need to make this more explicit here.